src/test/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/CalculationResultEnrichmentServiceTest.kt

```diff
@@ -0,0 +1,838 @@
+  }
+
+  @Test
+  fun `should add HDCED before PRRD hint if PRRD is present in other dates and is after HDCED`() {
+    val (hdcedDate, hdcedType) = getReleaseDateAndStubAdjustments(ReleaseDateType.HDCED, LocalDate.of(2021, 2, 1))
+
+    val calculationRequest = calculationRequest(listOf(calculationOutcome(hdcedType, hdcedDate)))
+    val calculationBreakdown = CalculationBreakdown(
+      emptyList(),
+      null,
+      emptyMap(),
+      mapOf(ReleaseDateType.PRRD to LocalDate.of(2021, 2, 2)),
+    )
+    val results = calculationResultEnrichmentService().addDetailToCalculationDates(calculationRequest, calculationBreakdown)
+    assertThat(results[hdcedType]?.hints).isEqualTo(listOf(ReleaseDateHint("Release on HDC must not take place before the PRRD Tuesday, 02 February 2021")))
+  }
+
+  @Test
+  fun `should not add HDCED before PRRD hint if PRRD is present in other dates but is before HDCED`() {
+    val (hdcedDate, hdcedType) = getReleaseDateAndStubAdjustments(ReleaseDateType.HDCED, LocalDate.of(2021, 2, 1))
+
+    val calculationRequest = calculationRequest(listOf(calculationOutcome(hdcedType, hdcedDate)))
+    val calculationBreakdown = CalculationBreakdown(
+      emptyList(),
+      null,
+      emptyMap(),
+      mapOf(ReleaseDateType.PRRD to LocalDate.of(2021, 2, 1)),
+    )
+    val results = calculationResultEnrichmentService().addDetailToCalculationDates(calculationRequest, calculationBreakdown)
+    assertThat(results[hdcedType]?.hints).isEqualTo(emptyList<ReleaseDateHint>())
+  }
+
+  @Test
+  fun `should add all HDCED hints if they all apply`() {
+    val (hdcedDate, hdcedType) = getReleaseDateAndStubAdjustments(ReleaseDateType.HDCED, LocalDate.of(2021, 2, 1))
+    val (mtdDate, mtdType) = getReleaseDateAndStubAdjustments(ReleaseDateType.MTD, hdcedDate.plusDays(1))
+
+    val sentencesAndOffences = listOf(
+      sentenceAndOffence("DTO"),
+      sentenceAndOffence("LR"),
+    )
+    val calculationRequest = calculationRequest(listOf(calculationOutcome(hdcedType, hdcedDate), calculationOutcome(mtdType, mtdDate)), sentencesAndOffences)
+    val calculationBreakdown = CalculationBreakdown(
+      emptyList(),
+      null,
+      mapOf(ReleaseDateType.HDCED to ReleaseDateCalculationBreakdown(setOf(CalculationRule.HDCED_ADJUSTED_TO_CONCURRENT_CONDITIONAL_RELEASE))),
+      mapOf(ReleaseDateType.PRRD to LocalDate.of(2021, 2, 2)),
+    )
+    val results = calculationResultEnrichmentService().addDetailToCalculationDates(calculationRequest, calculationBreakdown)
+    assertThat(results[hdcedType]?.hints).isEqualTo(
+      listOf(
+        ReleaseDateHint("HDCED adjusted for the CRD of a concurrent sentence or default term"),
+        ReleaseDateHint("Release on HDC must not take place before the PRRD Tuesday, 02 February 2021"),
+        ReleaseDateHint("The Detention and training order (DTO) release date is later than the Home detention curfew eligibility date (HDCED)"),
```

Comment:
I'm not sure what the functionality is here tbf, but I'm guessing this is just intended as a nudge as we don't currently try and implement the logic for DTO and HDC releases as they are fairly wordy - else i'd expect the top and bottom hint to be mutually exclusive

Reply:
that makes 2 of us but it was supported in the UI so I just wanted to make sure the if's were nested correctly. Dunno if half of these are real scenarios or not. They were generated by just looking at every branch of the UI code

Reply:
I'll chat with scott / cherky but yeah you've just ported across 👍🏻